### ISSUE TITLE
[New rule] new-api

### ISSUE THREAD

**Issue by ivk1800** [dart-code-checker/dart-code-metrics#879]
**Please describe what the rule should do:**
Avoid call a new api if language version constraints are violated.

**What category of rule is this? (place an "X" next to just one item)**

[x] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about (it will be better if you can provide both good and bad examples):**

BAD:
dart version in pubspec `sdk: ">=2.12.0 <3.0.0"`
installed dart version 2.17

```dart
unawaited(Future.value()); // LINT, because unawaited since 2.15
```

GOOD:
dart version in pubspec `sdk: ">=2.15.0 <3.0.0"`
installed dart version 2.17

```dart
unawaited(Future.value());
```

**Are you willing to submit a pull request to implement this rule?**
Yes

**Comment by incendial** [comment]
@ivk1800 hi, which package provides `unawaited`? Asking, since the analyzer has highlight for features that are used the way you described, there should be no need to create a rule for that

**Comment by ivk1800** [comment]
> @ivk1800 hi, which package provides `unawaited`
`dart:async` 
My analyzer doesn't show anything, maybe I don't have the rule enabled?

**Comment by incendial** [comment]
It's not a rule, try setting `sdk: ">=2.15.0 <3.0.0"` and use super parameters or Class.new and you'll see the highlight. The only question remaining is why it doesn't work for dart:async. Have you tried to ask the Dart team first?

**Comment by ivk1800** [comment]
> It's not a rule, try setting `sdk: ">=2.15.0 <3.0.0"` and use super parameters or Class.new and you'll see the highlight. The only question remaining is why it doesn't work for dart:async. Have you tried to ask the Dart team first?

analyzer has highlight `super parameters or Class.new` because it is feature of language. `unawaited `just  annotated with `@Since`, anything can be annotated by this, this is just an example. 

I did not ask the dart team why it does not work, but it looks strange. Look at the rule from android "NewApi" as reference.

**Comment by incendial** [comment]
I mean, the annotation comes from the core package, the `unawaited` comes from the core package, that's why I think that a rule (or even some built-in analyzer support) that highlights such cases should be in the core package and not in the DCM.

**Comment by incendial** [comment]
So that's why I suggest to start the discussion with the Dart team. If they refuse, we can add this to DCM.

**Comment by incendial** [comment]
Closing this. Please feel free to reopen, if you disagree or the Dart team does not want to add this rule
